src/test/scala/io/citrine/lolo/bags/BaggedResultTest.scala

```diff
@@ -31,6 +31,50 @@ class BaggedResultTest {
     }
   }
 
+  @Test
+  def testBaggedSingleResultGetUncertainty(): Unit = {
+    val noiseLevel = 100.0
+    val rng = new Random(237485L)
+    Seq(RegressionTreeLearner(),GuessTheMeanLearner()).foreach{ baseLearner =>
+      Seq(30,100,301).foreach { nRows =>
+        val trainingDataTmp = TestUtils.generateTrainingData(nRows, 1, noise = 0.0, function = _ => 0.0, seed = rng.nextLong())
+        val trainingData = (trainingDataTmp).map { x => (x._1, x._2 + noiseLevel * rng.nextDouble()) }
+        val baggedLearner = Bagger(baseLearner, numBags = 2 * nRows, uncertaintyCalibration = true)
+        val RFMeta = baggedLearner.train(trainingData)
+        val RF = RFMeta.getModel()
+
+        val results = RF.transform(trainingData.take(1).map(_._1))
+        val sigmaObs: Seq[Double] = results.getUncertainty().get.asInstanceOf[Seq[Double]]
+        val sigmaMean: Seq[Double] = results.getUncertainty(observational = false).get.asInstanceOf[Seq[Double]]
+
+        sigmaMean.zip(results.asInstanceOf[RegressionResult].getStdDevMean().get).foreach{ case (a,b) =>
+          assert(a == b, "Expected getUncertainty(observational=false)=getStdDevMean()")
+        }
+        sigmaObs.zip(results.asInstanceOf[RegressionResult].getStdDevObs().get).foreach{ case (a,b) =>
+          assert(a == b, "Expected getUncertainty()=getStdDevObs()")
+        }
+
+        sigmaObs.zip(sigmaMean).foreach { case (sObs, sMean) => assert(sObs > sMean, "Uncertainty should be greater when observational = true.") }
+
+        // We have strong theoretical guarantees on the behavior of GuessTheMeanLearner, so let's exercise them.
+        // TODO(grobinson): implement similar test for RegressionTreeLearner.
+        if (baseLearner.isInstanceOf[GuessTheMeanLearner]) {
+          // NOTE: these bounds reflect a ~3x systematic variance under-estimation in this particular test setting.
+          val rtolLower = 5.0  // Future recalibration should decrease this number.
+          val rtolUpper = 1.0  // Future recalibration should increase this number.
```

Comment:
Decrease? Or I think the assert statement should be changed to `s * rTolUpper < noiseLevel`. Currently increasing `rTolUpper` makes the assertion weaker

Reply:
Future recalibration should make both just slightly more than 1. It's set to 1 now to catch changes, because it's currently underestimating consistently, but a well-calibrated estimate would have both around, say, 2.

Reply:
OK, so `sigmaObs` is currently under-estimating, and `s < 1 * noiseLevel` is reliably true now, but if it were working correctly a more appropriate bound would be `s < 2 * noiseLevel`? And you're leaving it at 1 so you can observe the test "failing" when sigmaObs is re-calibrated, at which point you'll increase `rTolUpper`? I think that makes sense.